src/test/java/bio/terra/controller/DatasetOperationTest.java

```diff
@@ -0,0 +1,341 @@
+        ClassLoader classLoader = getClass().getClassLoader();
+        String datasetJson = IOUtils.toString(classLoader.getResourceAsStream("dataset-test-dataset-baddata.json"));
+        DatasetRequestModel badDataRequest = objectMapper.readerFor(DatasetRequestModel.class).readValue(datasetJson);
+        badDataRequest.getContents().get(0).getSource().setStudyName(studyRequest.getName());
+
+        String baseName = badDataRequest.getName();
+        String datasetName = randomizedName(baseName, "baddata");
+        badDataRequest.setName(datasetName);
+        String jsonRequest = objectMapper.writeValueAsString(badDataRequest);
+
+        MvcResult result = mvc.perform(post("/api/repository/v1/datasets")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRequest))
+// TODO: swagger field validation errors do not set content type; they log and return nothing
+//                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        MockHttpServletResponse response = validateJobModelAndWait(result);
+
+// TODO: the status code is always 201 right now and the response is the dataset summary model, because
+// jobs doesn't notice that the flight failed...
+        assertThat(response.getStatus(), equalTo(201));
+//        ErrorModel errorModel = objectMapper.readValue(response.getContentAsString(), ErrorModel.class);
+//        assertThat(errorModel.getMessage(), containsString("Fred"));
+    }
+
+    // !!! This test is intended to be run manually when the BigQuery project gets orphans in it.
+    // !!! It tries to delete all datasets from the project.
+    // You have to comment out the @Ignore to run it and not forget to uncomment it when you are done.
+    @Ignore
+    @Test
+    public void deleteAllBigQueryProjects() throws Exception {
+        // Collect a list of datasets. Then delete each one.
+        List<DatasetId> idList = new ArrayList<>();
+        for (com.google.cloud.bigquery.Dataset dataset :  bigQuery.listDatasets().iterateAll()) {
+            idList.add(dataset.getDatasetId());
+        }
+
+        for (DatasetId id : idList) {
+            bigQuery.delete(id, BigQuery.DatasetDeleteOption.deleteContents());
+        }
+    }
+
+    private DatasetSummaryModel createTestDataset(DatasetRequestModel datasetRequest, String infix) throws Exception {
+        String baseName = datasetRequest.getName();
+        String datasetName = randomizedName(baseName, infix);
+        datasetRequest.setName(datasetName);
+
+        String jsonRequest = objectMapper.writeValueAsString(datasetRequest);
+
+        MvcResult result = mvc.perform(post("/api/repository/v1/datasets")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(jsonRequest))
+// TODO: swagger field validation errors do not set content type; they log and return nothing
+//                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
```

Comment:
Can you assert that the response code is a 4xx where we expect validation errors? That's how I've been testing these validations.

Reply:
I don't expect a failure here. In developing the tests, I had errors in the input file that caused validation errors. But instead of seeing those errors, I was seeing this Expect fail.

I think we need to get the validation errors to produce proper ErrorModel. We can add a common failure check in the validateJobModel and wait.